ios/Classes/BetterPlayerPlugin.m

```diff
@@ -193,18 +193,40 @@ - (void) setupRemoteCommands:(BetterPlayer*)player isExtraVideo:(BOOL)isExtraVid
 }
 
 - (void) setupRemoteCommandNotification:(BetterPlayer*)player, NSString* title, NSString* author , NSString* imageUrl, BOOL isLiveStream {
-    float positionInSeconds = player.position /1000;
-    float durationInSeconds = player.duration/ 1000;
-
+    // This function is always called double times at the end of video due to the default behavior of AVPlayer
+    // This check is used to prevent the latest call.
+    if (player.position >= player.duration - 500) {
+        return;
```

Comment:
@noboru-i 
eg: Duration: `32.000000` (the duration always got round in all videos)
If the end position is `31.768002`
this process will be like this:
at `31.000000`: display `-0:01` and call the `dispatch_after(0.5)` 
at `31.500000`: the code block in `dispatch_after` got call and update the `position = duration`, and display `-0:00`
at `31.768002`: this function got call by system, the calculated position is still `31.000000` ==> the UI jump back to `-0:01`

Reply:
but if i use your suggestion:

> We may need F like 1000F.

at the `31.768002`: the control center will round up it to `32.000000` and display `-0:00`
but the progress bar still is `31.768002` absolutely, then we will see a very very small jump in progress bar because it got a `~0.3s` jump back (we can only see that jump in a very short video like duration ~ 30s, in other longer video it act like normal)

Reply:
@mvn-hienvu-hn 
Oh sorry... You are right.

I feel strange we cannot get the correct "duration".
(If we can get the correct "duration", we can detect the "end" by `position == duration`.)

Sorry. I want to confirm.

- https://github.com/dwango-nfc/betterplayer/blob/main/ios/Classes/BetterPlayer.m#L522 `[[_player currentItem] duration]` is also rounded?
- We also get rounded duration in [iOS sample code](https://developer.apple.com/documentation/mediaplayer/becoming_a_now_playable_app) ?

Reply:
@noboru-i 
There is what i got:

> https://github.com/dwango-nfc/betterplayer/blob/main/ios/Classes/BetterPlayer.m#L522 [[_player currentItem] duration] is also rounded?

Yes, it always get rounded.

> We also get rounded duration in [iOS sample code](https://developer.apple.com/documentation/mediaplayer/becoming_a_now_playable_app) ?

No.

I've tried using `CMTimeCompare([[_player currentItem] duration], [_player currentTime]`,
But in the both case, it always return `false`, those `CMTime` can't reach each other.

Reply:
Thanks. I understand.